spring-session/src/main/java/org/springframework/session/data/mongo/MongoOperationsSessionRepository.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.mongo;
+
+import com.mongodb.DBObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.IndexOperations;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.index.Index;
+import org.springframework.data.mongodb.core.index.IndexInfo;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.session.FindByIndexNameSessionRepository;
+
+import javax.annotation.PostConstruct;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Session repository implementation which stores sessions in Mongo.
+ * Uses {@link Converter} to transform session objects from/to
+ * native Mongo representation ({@code DBObject}).
+ *
+ * Repository is also responsible for removing expired sessions from database.
+ * Cleanup is done every minute.
+ *
+ * @author Jakub Kubrynski
+ * @since 1.2
+ */
+public class MongoOperationsSessionRepository implements FindByIndexNameSessionRepository<MongoExpiringSession> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(MongoOperationsSessionRepository.class);
+
+	public static final String EXPIRES_AT_FIELD = "expireAt";
```

Comment:
These to fields should likely be on a parent interface of  `StandardMongoSessionToDBObjectConverter` and `StandardDBObjectToMongoSessionConverter` since they only make sense if those `Converter`s are being used.